src/graph.cpp

```diff
@@ -229,4 +236,61 @@ std::vector<Vertex *> Graph::findEndOfLines(const int stationId) const {
     }
 
     return eol_stations;
-}
\ No newline at end of file
+}
+
+/**
+ * DFS traversal variation that sets the visited attribute to true of the nodes the DFS traverses to
+ * Time Complexity: O(|V|+|E|)
+ * @param source - Vertex where the DFS starts
+ */
+void Graph::visitedDFS(Vertex *source) {
+    source->setVisited(true);
+    for (Edge *e : source->getAdj()){
+        if (!e->getDest()->isVisited()){
+            visitedDFS(e->getDest());
+        }
+    }
+}
+
+/**
+ * Finds the pairs of stations with max Max-Flow
+ * Time Complexity: O(V^2*(VE^2)) //TODO someone check if V^2 is the correct ammount of times EdmondsKarp is called
+ * @return
+ */
+std::pair<std::list<std::pair<Vertex *,Vertex *>>,unsigned int> Graph::calculateNetworkMaxFlow() {
+    unsigned int max = -1;
+    std::list<std::pair<Vertex *,Vertex *>> stationList;
+    for (auto itV1 = vertexSet.begin(); itV1 < vertexSet.end(); itV1++){
+        for (auto itV2 = itV1; itV2 < vertexSet.end(); itV2++){
+            Vertex * v1 = *itV1++;
+            Vertex * v2 = *itV2;
+            for (Vertex *aux : vertexSet) aux->setVisited(false);
+            visitedDFS(v1);
+            if (v2->isVisited()){
+                unsigned int itFlow = edmondsKarp({v1->getId()}, v2->getId());
+                if (itFlow == max) stationList.push_back({v1,v2});
+                if (itFlow > max) {
+                    max = itFlow;
+                    stationList = {{v1, v2}};
+                }
+            }
+        }
+    }
+    return {stationList, max};
+}
+
+/**
+ * Finds the incoming flux that a certain station can receive (i.e the ammount of trains that can arrive there at the same time)
+ * Time Complexity: O(|VE^2|)
+ * @param station - Station's ID
+ */
+unsigned int Graph::incomingFlux(const std::string &station) {
+    std::list<std::string> super_source = findEndOfLines(station);
+    for (auto it = super_source.begin(); it != super_source.end(); it++)
```

Comment:
Why go through all the trouble verifying if the station is the list if the worst that can happens is edmonds-karp finds a 0 flow path between the vertex and itself? It will not interfere if there are higher flows. If you really think it's important, you can instead verify the size of station.adj to see if it will be considered an end of line

Reply:
The problem isn't there, the problem will be during the bfs. I kept this part here to simplify, but I can refactor it to the path() method, even though will add some lines than already has

Reply:
The way I see it, if the source node is already in the queue, it will analyze its own adjacent edges and not "find itself", so the function wont end and it will keep going through all the other nodes in the queue until it can find a path, and if not, just return false. Maybe you can verify this with a simple test case.